### PR TITLE
TS-7: Refactoring routing element

### LINKED ISSUE #7: Refactoring routing element
Create router components for roles **_"users, manager and teacher"_**. Make them in separate files that will test the user's role. Also, refactor NavigationPage component.

Estimate:  1.5 days
Priority: _High_

### PR DESCRIPTION
**Pull request by HiroFive**
Fixed route in app and refactored NavigationPage component

### CODE REVIEW

File: frontend/public/assets/i18n/translations/uk/common.json [hunk 1]
```diff
@@ -174,5 +174,6 @@
   "one_day": "Один день",
   "preview": "Переглянути",
   "vacation_label": "Вихідний",
-  "regular_lesson_label": "Пара за розкладом"
+  "regular_lesson_label": "Пара за розкладом",
+  "more_label": "Бліьше"
```
Comment (ViktoriiaKosovan) [inline frontend/public/assets/i18n/translations/uk/common.json]: wrong word "Більше"

File: frontend/src/containers/AdminPage/AdminPage.js [hunk 2]
```diff
@@ -1,16 +1,9 @@
...
-    return (
-        <>
-            <NavigationPage val={navigation.LESSONS} />
-            <LessonPage />
-        </>
-    );
+    return <LessonPage />;
 }
 const mapStateToProps = (state) => ({
     classScheduler: state.classActions.classScheduler,
```
Comment (Akr1s) [inline frontend/src/containers/AdminPage/AdminPage.js]: Why do we use AdminPage as a wrapper here? Also, mapStateToProps is unnecessary

File: frontend/src/router/AdminRouter.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+                <AdminRoute path={links.MergeRolePage} component={MergeRolePage} />
+                <AdminRoute path={links.Changes} component={TemporarySchedule} />
+
+                <AdminRoute path={links.ACTIVATION_PAGE}>
+                    <Redirect to={links.ADMIN_PAGE} />
+                </AdminRoute>
+                <AdminRoute path="/">
+                    <Redirect to={links.LessonPage} />
+                </AdminRoute>
+            </Switch>
```
Comment (Akr1s) [inline frontend/src/router/AdminRouter.js]: I think we should consider using React.lazy in such cases

File: frontend/src/components/AddGroupForm/AddGroupForm.js [hunk 4]
```diff
@@ -21,10 +23,15 @@ import {
...
-    const { handleSubmit, pristine, onReset, submitting, match, group, initialize } = props;
+    const location = useLocation();
+    const { handleSubmit, pristine, onReset, submitting, group, initialize } = props;
 
     useEffect(() => {
-        if (group && match.url.includes(links.Edit) && !match.url.includes(links.Student)) {
+        if (
+            group &&
+            location.pathname.includes(links.Edit) &&
+            !location.pathname.includes(links.Student)
```
Comment (Niknikson) [inline frontend/src/components/AddGroupForm/AddGroupForm.js]: we will have  to check all pages if url includes edit, delete... . Can we create a helper for this ?

File: frontend/src/components/Navigation/NavigationPanel.scss [hunk 5]
```diff
@@ -0,0 +1,63 @@
+.admin-navigation {
```
Comment (andrii-kuzmichov) [inline frontend/src/components/Navigation/NavigationPanel.scss:4]: create or reuse constants for colors

File: frontend/src/constants/links.js [hunk 6]
```diff
@@ -1,15 +1,15 @@
 const links = {
```
Comment (andrii-kuzmichov) [inline frontend/src/constants/links.js]: maybe refactor this file separate key links and reuse it for secondary&

File: frontend/src/constants/navigation.js [hunk 7]
```diff
@@ -1,27 +1,28 @@
...
+    ROOM_LIST: 'RoomList_management_title',
+    SUBJECT_PAGE: 'SubjectPage_management_title',
+    SEMESTER_PAGE: 'SemesterPage_management_title',
+    DEPARTMENTS: 'Departments_management_title',
+    LESSON_PAGE: 'LessonPage_management_title',
+    MERGE_ROLE_PAGE: 'MergeRolePage_management_title',
+    CHANGES: 'Changes_management_title',
+    BUSY_ROOMS: 'BusyRooms_management_title',
+    GENERAL: 'General_management_title',
 };
```
Comment (andrii-kuzmichov) [inline frontend/src/constants/navigation.js]: Why we need it?

Comment (HiroFive) [reply frontend/src/constants/navigation.js]: Done, now I'm using common constants

File: frontend/src/components/Navigation/NavigationPanel.scss (hunk 5 above)
Comment (HiroFive) [reply frontend/src/components/Navigation/NavigationPanel.scss:4]: Done

File: frontend/src/constants/links.js (hunk 6 above)
Comment (HiroFive) [reply frontend/src/constants/links.js]: Replaced by individual links

File: frontend/src/containers/GroupList/GroupList.scss [hunk 8]
```diff
@@ -11,6 +11,14 @@ $group-size: 10px;
    font-size: 20px;
  }
 }
+.link-href{
+  text-decoration: none;
+  color: black;
+  &:hover{
+    color: black;
```
Comment (ViktoriiaKosovan) [inline frontend/src/containers/GroupList/GroupList.scss]: Could you add this color to color constants too?

File: frontend/public/assets/i18n/translations/uk/common.json (hunk 1 above)
Comment (HiroFive) [reply frontend/public/assets/i18n/translations/uk/common.json]: fixed

File: frontend/src/containers/GroupList/GroupList.scss (hunk 8 above)
Comment (HiroFive) [reply frontend/src/containers/GroupList/GroupList.scss]: add to constants

**Review by Niknikson — APPROVED** [review]

**Review by Akr1s — APPROVED** [review]

**Pull request closed without merging**